tests/5.1/begin_end_declare_variant.c

```diff
@@ -0,0 +1,62 @@
+//===---- test_declare_target_nested.c ---------------------------------------------===//
```

Comment:
We will need to change the header to reflect name of the new test and also change the description of the test. (should also remove @Todo)